Lib/fontTools/pens/hashPointPen.py

```diff
@@ -33,8 +53,11 @@ class HashPointPen(AbstractPointPen):
     >    pass
     """
 
-    def __init__(self, glyphWidth=0, glyphSet=None):
+    def __init__(
+        self, glyphWidth=0, glyphSet=None, roundFunction: Callable | None = None
+    ):
         self.glyphset = glyphSet
+        self.roundFunction = roundFunction
```

Comment:
I find a bit strange that immediately below this there is a call to built-in `round` to round the advance width, a user might immediately wonder why not use the roundFunction that I just provided?

Reply:
if this function is only meant to round the components' 2x2 transform, then it should be renamed accordingly, because as it stands it suggests that it's going to be used to round in general which is not the case (advance width and glyph points or translation dx,dy are not being rounded).

I think overall that this HashPointPen could stay as it is and not round anything, and instead you create a separate FilterPen subclass that only does what you want, that is round the component's transform as desired.

Reply:
we have already a RoundingPen and RoundingPointPen here:
https://github.com/fonttools/fonttools/blob/306f40a2aea237ca98d2d207847164c6784bc79f/Lib/fontTools/pens/roundingPen.py#L9

these only apply the rounding to point coords and dx/dy offsets, but maybe you could add an additional parameter to round the component transform floats.
Then you could use this in conjunction to the HashPointPen to achieve the desired effect

Reply:
I just found another complication: the HashPointPen does not conform [to the UFO spec](https://github.com/unified-font-object/ufo-spec/issues/224) at all anymore ...

Reply:
Alright, I forgot about RoundingPointPen.

I think calculating a hash to compare source data to TTF data is a common use case, so maybe there should be a readymade solution for it? (I thought that ufo2ft didn't compare the UFO hash against TTF glyphs, but actually it does) So maybe I could add another pen that combines HashPointPen and RoundingPointPen to make comparing with TTF easy?

Reply:
first of all we could add the optional `transformRoundFunction` parameter to the rounding pens, that defaults to `noRound`. Then you could already start using them in combination with the HashPointPen in ufo2ft or wherever you need them. It's just one additional line to construct the combined pen, I don't think we need to add anything here. You could add to the docstring of HashPointPen an example.

Reply:
Ok, I'll start a new PR for that.